axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/bankreconciliation/BankReconciliationService.java

```diff
@@ -268,12 +274,13 @@ protected MoveLine generateMoveLine(
     String description = "";
     description =
         description.concat(bankReconciliationLine.getBankStatementLine().getDescription());
-    if (description.length() > DESCRIPTION_SIZE_LIMIT) {
-      description = description.substring(0, DESCRIPTION_SIZE_LIMIT - 1);
-    }
-    description = description.concat("ref:");
+    description = StringTool.cutTooLongString(description);
+
     if (!Strings.isNullOrEmpty(bankReconciliationLine.getReference())) {
-      description = description.concat(bankReconciliationLine.getReference());
+      String reference = "ref:";
+      reference = reference.concat(bankReconciliationLine.getReference());
+      description = StringTool.cutTooLongStringWithOffset(description, reference.length());
+      description = description.concat(reference);
```

Comment:
I think it's better to concat all values and then limit the result to the target field length.

No need to add a new method.